### ISSUE TITLE
Channel descriptions are not saved by write.FCS

### ISSUE THREAD

**Issue by yannabraham** [RGLab/flowCore#98]
I'm generating flowFrames through the flowFrame() function, manually editing the channel descriptions through the parameters() function, changing the desc column. When I save those flowFrames using write.FCS(), then load them back using read.FCS() the channel description is empty. It looks like the description is read from the keywords, namely the $P00S keywords, which are not filled when the FCS file is saved. 

I believe this is an unexpected behavior, that should be either documented or fixed.

**Comment by SamGG** [comment]
I am not a developer. Your proposal sounds interesting. Concerning the documentation, just remember that write.FCS is stated as "experimental". Meanwhile I guess you easily solved this issue by filling the keywords before write.FCS.

**Comment by mikejiang** [comment]
Try the fix.

**Comment by SamGG** [comment]
Hi Mike.
I thought the fix was pushed to bioconductor, but it was not there. So, I looked at your fix in order to get the interesting part without installing the devel code of flowCore. I would like to suggest you a few points.
* `pns <- pd[["desc"]]`: the desc column may be a factor leading to number instead of text.
* `sprintf("$P%sS", newid.pns)`: technically newid.pns is an integer so the format should use %d instead of %s, but due to R cast-magic %s works.
* `mk <- c(mk, pns)`: in fact pns have already been pushed into the new description list `mk`, so pns are duplicated; `description(x) <- mk` manages this duplication, but I don't know how priority takes place.

Anyway, I solved my issue, and I hope you will have time to update this and clean the code before the next release. Thanks for your work at flowCore.
